### PR TITLE
DownloadGraph: Hide chart points from before a version was released

### PR DESCRIPTION
**Pull request by Turbo87**
Resolves https://github.com/rust-lang/crates.io/issues/3305

Example for the `rand` crate:

Bildschirmfoto 2021-03-10 um 15 32 34

### CODE REVIEW

**Comment by Turbo87** [comment]
and don't be scared by the long diff. the majority is a testing fixture... 😅

File: app/components/download-graph.js [hunk 1]
```diff
@@ -67,81 +75,109 @@ export default class DownloadGraph extends Component {
...
-      versions.add(version_num);
+    let version_num = version.num;
 
-      let key = d.date;
-      if (dates[key]) {
-        let prev = dates[key].cnt[version_num] || 0;
-        dates[key].cnt[version_num] = prev + d.downloads;
-      }
-    });
+    versions.set(version_num, version);
```
Comment (pichfl) [inline app/components/download-graph.js:105]: Is there a reason this uses underscored variables over camelCase like most of the other code here?

Comment (Turbo87) [reply app/components/download-graph.js:105]: only legacy reasons 😉

**Comment by Turbo87** [comment]
Fixed CI by using https://github.com/ozkxr/match-json for the download data assertion to make it possible to use predicate functions when comparing JSON objects. :)

**Pull request merged**